src/core/system_impl.cpp

```diff
@@ -25,7 +25,8 @@ SystemImpl::SystemImpl(MavsdkImpl& parent, uint8_t system_id, uint8_t comp_id, b
     _receive_commands(*this),
     _timesync(*this),
     _ping(*this),
-    _mission_transfer(*this, _message_handler, _parent.timeout_handler)
+    _mission_transfer(
+        *this, _message_handler, _parent.timeout_handler, [this]() { return timeout_s(); })
```

Comment:
I am not sure I understand why there is this `TimeoutSCallback` callback at all, since it just returns the value of the `timeout_s` member? :thinking:

Reply:
I want to get the latest timeout set by the user by the time a transfer is initiated. As the MissionTransfer instance is created with the system we might miss the user updating the timeout. Therefore, I've added the callback to get the value later. The alternative would be to pass the timeout every time you do a transfer, so then the timeout value would have to be fed through via the plugin which is using the MissionTransfer.
I guess the problem there is that the plugin is not talking to the SystemImpl API but directly getting the MissionTransfer class here:
https://github.com/mavlink/MAVSDK/blob/develop/src/core/system_impl.h#L232